src/EditorFeatures/VisualBasicTest/AutomaticEndConstructCorrection/AbstractCorrectorTests.vb

```diff
@@ -0,0 +1,113 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Utilities
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Editor.VisualBasic.AutomaticEndConstructCorrection
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.Text.Shared.Extensions
+Imports Microsoft.VisualStudio.Text
+
+Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.AutomaticEndConstructCorrection
+    Public MustInherit Class AbstractCorrectorTests
+        Friend MustOverride Function CreateCorrector(buffer As ITextBuffer, waitIndicator As TestWaitIndicator) As ICorrector
+
+        Protected Sub VerifyContinuousEdits(codeWithMarker As String,
+                                      type As String,
+                                      expectedStringGetter As Func(Of String, String),
+                                      removeOriginalContent As Boolean,
+                                      Optional split As String = Nothing)
+            ' do this since xml value put only vbLf
+            codeWithMarker = codeWithMarker.Replace(vbLf, vbCrLf)
```

Comment:
Ancient convention was to never allow these strings in. Pass it through a the XML Element directly, or do .NormalizedValue on each of the call sites where you do the conversion.

("Ancient convention" in this case was me a few years ago trying to make this sane. We had so many helpers converting back and forth that I couldn't figure out what the heck was going on. So I tried to enforce "strings always have CRLF" and was more careful at the conversion points.)